streaming/vision/convert/coco.py

```diff
@@ -0,0 +1,231 @@
+
+    def __len__(self):
+        return len(self.images)
+
+    def __getitem__(self, idx: int):
+        img_id = self.img_keys[idx]
+        img_data = self.images[img_id]
+        fn = img_data[0]
+        img_path = os.path.join(self.img_folder, fn)
+
+        img = Image.open(img_path).convert('RGB')
+
+        htot, wtot = img_data[1]
+        bbox_sizes = []
+        bbox_labels = []
+
+        for (l, t, w, h), bbox_label in img_data[2]:
+            r = l + w
+            b = t + h
+            bbox_size = (l / wtot, t / htot, r / wtot, b / htot)
+            bbox_sizes.append(bbox_size)
+            bbox_labels.append(bbox_label)
+
+        bbox_sizes = torch.tensor(bbox_sizes)
+        bbox_labels = torch.tensor(bbox_labels)
+
+        return img, img_id, (htot, wtot), bbox_sizes, bbox_labels
+
+
+def parse_args() -> Namespace:
+    """Parse command line arguments.
+
+    Args:
+        Namespace: Command line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--in_root',
+                      type=str,
+                      default='./datasets/coco/',
+                      help='Location of Input dataset. Default: ./datasets/coco/')
+    args.add_argument(
+        '--out_root',
+        type=str,
+        default='./datasets/mds/coco/',
+        help='Location to store the compressed dataset. Default: ./datasets/mds/coco/')
+    args.add_argument('--splits',
+                      type=str,
+                      default='train,val',
+                      help='Split to use. Default: train,val')
+    args.add_argument('--compression',
+                      type=str,
+                      default='zstd:7',
+                      help='Compression algorithm to use. Default: zstd:7')
+    args.add_argument('--hashes',
+                      type=str,
+                      default='sha1,xxh64',
+                      help='Hashing algorithms to apply to shard files. Default: sha1,xxh64')
+    args.add_argument(
+        '--limit',
+        type=int,
+        default=1 << 25,
+        help='Shard size limit, after which point to start a new shard. Default: 33554432')
+    args.add_argument('--progbar',
+                      type=int,
+                      default=1,
+                      help='tqdm progress bar. Default: 1 (Act as True)')
+    args.add_argument(
+        '--leave',
+        type=int,
+        default=0,
+        help=
+        'Keeps all traces of the progressbar upon termination of iteration. Default: 0 (Act as False)'
+    )
+    return args.parse_args()
+
+
+def each(dataset: _COCODetection, shuffle: bool) -> Iterable[Dict[str, bytes]]:
+    """Generator over each dataset sample.
+
+    Args:
+        dataset (COCODetection): COCO detection dataset.
+        shuffle (bool): Whether to shuffle the samples.
+
+    Yields:
+        Sample dicts.
+    """
+    if shuffle:
+        indices = np.random.permutation(len(dataset))
+    else:
+        indices = list(range(len(dataset)))
```

Comment:
might as well use np.arange here

Reply:
Yes. Changed in https://github.com/mosaicml/streaming/pull/5/commits/e1eaca7428674cb2941960d1ad7c39c7091c1b12.